packages/core/src/receiver.ts

```diff
@@ -146,16 +146,46 @@ export function createRemoteReceiver(): RemoteReceiver {
         emit('mount');
       });
     },
-    insertChild: (id, index, child) => {
+    insertChild: (id, index, child, existingId) => {
       const attached = attachedNodes.get(
         id ?? ROOT_ID,
       ) as RemoteReceiverAttachableRoot;
+      const {children} = attached;
 
-      const normalizedChild = normalizeNode(child, addVersion);
-      retain(normalizedChild);
-      attach(normalizedChild);
+      let existingAttached: RemoteReceiverAttachableRoot | undefined;
+      let normalizedChild: RemoteReceiverAttachableChild;
 
-      const {children} = attached;
+      if (id === existingId) {
+        existingAttached = attached;
+      } else if (existingId !== false) {
+        existingAttached = attachedNodes.get(
+          existingId ?? ROOT_ID,
+        ) as RemoteReceiverAttachableRoot;
+      }
+
+      if (existingAttached) {
+        const childId = child.id;
+        const existingChildren = existingAttached.children;
+        const existingIndex = existingChildren.findIndex(
+          (child) => child.id === childId,
+        );
+
+        const [removed] = existingChildren.splice(existingIndex, 1);
+
+        normalizedChild = removed;
+
+        // If we are just moving the child to a different index in the same node,
+        // we don’t need to enqueue an update, because that will be done for this
+        // node below.
+        if (id !== existingId) {
+          existingAttached.version += 1;
+          enqueueUpdate(existingAttached);
+        }
```

Comment:
I don't see any test failures when commenting this out. Maybe there is a way to cover it?